src/arch/riscv/RiscvInterrupts.py

```diff
@@ -27,10 +28,35 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+from m5.citations import add_citation
 from m5.objects.BaseInterrupts import BaseInterrupts
+from m5.objects.IntPin import VectorIntSinkPin
+from m5.params import VectorParam
 
 
 class RiscvInterrupts(BaseInterrupts):
     type = "RiscvInterrupts"
     cxx_class = "gem5::RiscvISA::Interrupts"
     cxx_header = "arch/riscv/interrupts.hh"
+
+    local_interrupt_pins = VectorIntSinkPin("Pins for local interrupts")
```

Comment:
Just curious. Why we need to add `intSinkPin` in the `RiscvInterrupts`.

Don't you think post interrupt in the model without local_interrupt_pins connection  
`tc->getCpuPtr()->postInterrupt(tc->threadId(), <int_local_num>, 0)` is good?

Reply:
I think it would be better to modularize this, as it is more flexible. If you, for example, want to connect your device to an other hart, you just have to do a simple change in the configuration switch. If I want to post the interrupt directly from the external device model, you would have to count your hardware threads in order to obtain the correct thread.

Reply:
Ack